### PR TITLE
Migrate to `kubebuilder` and CRD v1, also drop `bazel`

### PR DESCRIPTION
**Pull request by Shareed2k**

### CODE REVIEW

File: controllers/redis_controller.go [hunk 1]
```diff
@@ -53,86 +47,33 @@ var (
...
-}
-
-// ReconcileRedis reconciles a Redis object
-type ReconcileRedis struct {
-	// This client, initialized using mgr.Client() above, is a split client
-	// that reads objects from the cache and writes to the apiserver
-	client client.Client
-	scheme *runtime.Scheme
+// RedisReconcile reconciles a Redis object
+type RedisReconcile struct {
```
Comment (shamil) [inline controllers/redis_controller.go:51]: Maybe rename it to `RedisReconciler`?

File: example/k8s_v1alpha1_redis_cr.yaml [hunk 2]
```diff
@@ -1,135 +1,53 @@
...
-  # and as the infix or suffix for them.
-  name: example
-spec:
-  # required field. Minimum value is 3
+    app.kubernetes.io/instance: shareed2k
+    app.kubernetes.io/name: redis
+  name: shareed2k
+  namespace: redis
+status:
+  master: redis-api-gateway-1
```
Comment (shamil) [inline example/k8s_v1alpha1_redis_cr.yaml:10]: isn't status is a read only field?

File: example/k8s_v1alpha1_redis_cr.yaml [hunk 3]
```diff
@@ -1,135 +1,53 @@
...
-  # the .metadata.labels will be added to all generated resources along with the
-  # redis=example label
   labels:
-    lol: woot
-  # the name of the Redis resource will be used as a label for all generated resources,
-  # and as the infix or suffix for them.
-  name: example
-spec:
-  # required field. Minimum value is 3
+    app.kubernetes.io/instance: shareed2k
```
Comment (shamil) [inline example/k8s_v1alpha1_redis_cr.yaml:5]: change to `example`

File: example/k8s_v1alpha1_redis_cr.yaml [hunk 4]
```diff
@@ -1,135 +1,53 @@
...
+    app.kubernetes.io/name: redis
+  name: shareed2k
+  namespace: redis
+status:
+  master: redis-api-gateway-1
   replicas: 3
-
-  # config is a set of key-value pairs needed for configuring Redis instances. (optional)
```
Comment (shamil) [inline example/k8s_v1alpha1_redis_cr.yaml:16]: this file containes useful docs, I would not remove them

**Review by shamil — CHANGES_REQUESTED** [review]

File: controllers/deepcontains.go [hunk 5]
```diff
@@ -29,7 +29,7 @@ type visit struct {
 	typ reflect.Type
 }
 
-// deepContains reports whether y is a ``subset'' of x defined as follows.
+// deepContains reports whether y is a “subset” of x defined as follows.
```
Comment (HagaiBarel) [inline controllers/deepcontains.go]: replace with `""`

File: controllers/redis_controller.go [hunk 6]
```diff
@@ -325,62 +266,74 @@ podIter:
...
+		return ctrl.Result{}, fmt.Errorf("failed to update Redis status: %s", err)
 	}
 	logger.Info("Updated Redis status")
-	return reconcile.Result{}, nil
+	return ctrl.Result{}, nil
 }
 
 // createOrUpdate abstracts away keeping in sync the desired and actual state of Kubernetes objects.
-// passing an empty instance implementing runtime.Object will generate the appropriate ``expected'' object,
+// passing an empty instance implementing runtime.Object will generate the appropriate “expected” object,
```
Comment (HagaiBarel) [inline controllers/redis_controller.go]: change to `""`

File: example/k8s_v1alpha1_redis_cr.yaml [hunk 7]
```diff
@@ -1,135 +1,53 @@
...
+    prometheus.io/port: '9121'
+    prometheus.io/scrape: 'true'
+  config:
+    client-output-buffer-limit: slave 0 0 0
+    maxmemory: 256mb
+    maxmemory-policy: volatile-ttl
+    repl-backlog-size: 64mb
+    save: 600 1
+  exporter:
+    image: oliver006/redis_exporter:v1.28.0
```
Comment (HagaiBarel) [inline example/k8s_v1alpha1_redis_cr.yaml:81]: There's a newer version available

**Review by HagaiBarel — CHANGES_REQUESTED** [review]

**Review by shamil — APPROVED** [review]

File: hack/boilerplate.go.txt [hunk 8]
```diff
@@ -1,4 +1,4 @@
-// Copyright YEAR The redis-operator Authors
+// Copyright 2019 The redis-operator Authors
```
Comment (HagaiBarel) [inline hack/boilerplate.go.txt:1]:
```suggestion
// Copyright 2022 The redis-operator Authors
```

**Review by HagaiBarel — APPROVED** [review]

File: PROJECT [hunk 9]
```diff
@@ -0,0 +1,16 @@
+domain: amaiz.com
+layout:
+- go.kubebuilder.io/v4-alpha
```
Comment (shamil) [inline PROJECT:3]: the stable one is `v3`, I guess it's ok to use `v4-alpha`

**Pull request merged**